integration-tests/src/tests/client/features/congestion_control.rs

```diff
@@ -0,0 +1,228 @@
+use near_chain_configs::Genesis;
+use near_client::test_utils::TestEnv;
+use near_client::ProcessTxResponse;
+use near_crypto::{InMemorySigner, KeyType, PublicKey};
+use near_primitives::account::id::AccountId;
+use near_primitives::errors::{ActionErrorKind, FunctionCallError, TxExecutionError};
+use near_primitives::shard_layout::ShardLayout;
+use near_primitives::transaction::SignedTransaction;
+use near_primitives::version::{ProtocolFeature, PROTOCOL_VERSION};
+use near_primitives::views::FinalExecutionStatus;
+use nearcore::test_utils::TestEnvNightshadeSetupExt;
+
+fn setup_runtime(sender_id: AccountId) -> TestEnv {
+    let epoch_length = 10;
+    let mut genesis = Genesis::test_sharded_new_version(vec![sender_id], 1, vec![1, 1, 1, 1]);
+    genesis.config.epoch_length = epoch_length;
+    genesis.config.protocol_version = ProtocolFeature::CongestionControl.protocol_version() - 1;
+    // Chain must be sharded to test cross-shard congestion control.
+    genesis.config.shard_layout = ShardLayout::v1_test();
+    TestEnv::builder(&genesis.config).nightshade_runtimes(&genesis).build()
+}
+
+/// Simplest possible upgrade to new protocol with congestion control enabled,
+/// no traffic at all.
+#[test]
+fn test_protocol_upgrade() {
+    // The following only makes sense to test is the feature is enabled in the current build.
+    if !ProtocolFeature::CongestionControl.enabled(PROTOCOL_VERSION) {
+        return;
+    }
+
+    // The immediate protocol upgrade needs to be set for this test to pass in
+    // the release branch where the protocol upgrade date is set.
+    std::env::set_var("NEAR_TESTS_IMMEDIATE_PROTOCOL_UPGRADE", "1");
```

Comment:
That is strange, the default should be immediate. This env variable should only be needed on the release branch where the voting date is scheduled in the future. What happens if you remove it? Are you by any chance based on a release branch or do you have the voting date set? You can grep for PROTOCOL_UPGRADE_SCHEDULE to check that.

Reply:
Ah, no I just copy-pasted this from similar tests. I thought we should set it already to not break things on the release branch later.
I think I can just remove it and CI is still green.